core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationLimitTests.java

```diff
@@ -0,0 +1,220 @@
+        transportServiceB = new TransportService(transportB, threadPool).start();
+        transportServiceB.acceptIncomingRequests();
+        nodeB = new DiscoveryNode("nodeB", transportServiceA.boundAddress().publishAddress(),
+            emptyMap(), emptySet(), Version.CURRENT);
+
+        transportA.connectToNode(nodeB);
+        clusterService = null;
+
+        clusterService = new ClusterService(settings, null, clusterSettings, threadPool, clusterName);
+        clusterService.setLocalNode(nodeA);
+        clusterService.setNodeConnectionsService(new NodeConnectionsService(Settings.EMPTY, null, null) {
+            @Override
+            public void connectToAddedNodes(ClusterChangedEvent event) {
+                // skip
+            }
+
+            @Override
+            public void disconnectFromRemovedNodes(ClusterChangedEvent event) {
+                // skip
+            }
+        });
+        clusterService.setClusterStatePublisher((event, ackListener) -> {
+        });
+        clusterService.start();
+    }
+
+    public void tearDown() throws Exception {
+        super.tearDown();
+        transportA.disconnectFromNode(nodeB);
+        transportServiceA.close();
+        transportServiceB.close();
+        clusterService.close();
+        terminate(threadPool);
+    }
+
+    public void testTransportReplicaActionNotSizeLimited() throws Exception {
+        final AtomicInteger count = new AtomicInteger(0);
+        final AtomicBoolean isRelocated = new AtomicBoolean(false);
+        String actionName = "testAction";
+        Action action = new Action(Settings.EMPTY, actionName, transportServiceA, clusterService, threadPool, count, isRelocated);
+
+        CountDownLatch replicaActionLatch = new CountDownLatch(1);
+        final AtomicReference<TransportException> capturedException = new AtomicReference<>();
+        final AtomicReference<TransportResponse> capturedResponse = new AtomicReference<>();
+        ShardId shard = new ShardId("test", "596D792A-0B84-11E6-9AF4-C738FDC8A409", 1);
+        transportServiceA.sendRequest(nodeB, action.transportReplicaAction, new Action.Request(shard), new
+            TransportResponseHandler<TransportResponse>() {
+            @Override
+            public TransportResponse newInstance() {
+                return new TransportResponse.Empty();
+            }
+
+            @Override
+            public void handleResponse(TransportResponse response) {
+                capturedResponse.set(response);
+                replicaActionLatch.countDown();
+            }
+
+            @Override
+            public void handleException(TransportException exp) {
+                capturedException.set(exp);
+                replicaActionLatch.countDown();
+            }
+
+            @Override
+            public String executor() {
+                return ThreadPool.Names.SAME;
+            }
+        });
+        replicaActionLatch.await(5, TimeUnit.SECONDS);
```

Comment:
I think we should just wait indefinitely and let the test time out if the latch is never tripped?